contrib/cli/great_expectations_contrib/types.py

```diff
@@ -0,0 +1,228 @@
+    CONCEPT_ONLY = "CONCEPT_ONLY"
+    EXPERIMENTAL = "EXPERIMENTAL"
+    BETA = "BETA"
+    PRODUCTION = "PRODUCTION"
+
+
+@dataclass(frozen=True)
+class GreatExpectationsContribPackage:
+    # Core
+    package_name: Optional[str] = None
+    icon: Optional[str] = None
+    description: Optional[str] = None
+    expectations: Optional[List[RenderedExpectationDetail]] = None
+    expectation_count: Optional[int] = None
+    dependencies: Optional[List[Dependency]] = None
+    maturity: Optional[Maturity] = None
+    status: Optional[PackageCompletenessStatus] = None
+
+    # Users
+    owners: Optional[List[GitHubUser]] = None
+    contributors: Optional[List[GitHubUser]] = None
+    domain_experts: Optional[List[DomainExpert]] = None
+
+    # Metadata
+    version: Optional[str] = None
+
+    # Existing configuration
+    config_file: ClassVar[str] = ".great_expectations_package.json"
+
+    def _to_json_file(self) -> None:
+        json_dict = self._to_json_dict()
+        data = json.dumps(json_dict, indent=4)
+        with open(self.config_file, "w") as f:
+            f.write(data)
+            logger.info(f"Succesfully wrote state to {self.config_file}.")
+
+    def _to_json_dict(self) -> Dict[str, Any]:
+        # Convert to JSON and remove nulls
+        json_dict = asdict(self)
+        to_delete = [key for key, val in json_dict.items() if val is None]
+        for key in to_delete:
+            del json_dict[key]
+        return json_dict
+
+    @classmethod
+    def from_json_file(cls) -> "GreatExpectationsContribPackage":
+        """Reads data from JSON file to create class instance.
+
+        If the target file is not available, an empty package is created.
+
+        Returns:
+            An instance of GreatExpectationsContribPackage to represent the package's current state
+
+        """
+        # If config file isn't found, create a blank JSON and write to disk
+        if not os.path.exists(cls.config_file):
+            instance = cls()
+            logger.debug(
+                f"Could not find existing package JSON; instantiated a new one"
+            )
+            return instance
+
+        with open(cls.config_file) as f:
+            contents = f.read()
+
+        data = json.loads(contents)
+        logger.info(f"Succesfully read existing package data from {cls.config_file}")
+        return cls(**data)
+
+    def update_package_state(self) -> None:
+        """
+        Parses diagnostic reports from package Expectations and uses them to update JSON state
+        """
+        diagnostics = self._retrieve_package_expectations_diagnostics()
+        self._update_attrs_with_diagnostics(diagnostics)
+        self._to_json_file()
+
+    def _update_attrs_with_diagnostics(self, diagnostics: List[Diagnostics]) -> None:
+        # TODO: Write logic to assign values to attrs
+        # This is a black box for now (don't have the full context just yet)
+        for diagnostic in diagnostics:
+            pass
```

Comment:
This will be implemented in a subsequent PR - this is setting up the foundation for the actual parsing of diagnostics.